internalsdk/auth/auth.go

```diff
@@ -51,35 +49,39 @@ type AuthClient interface {
 	Healthz(ctx context.Context) (bool, error)
 }
 
-// NewClient creates a new instance of AuthClient
-func NewClient(baseURL string, userConfig func() common.UserConfig) AuthClient {
-	// The default http.RoundTripper is ChainedNonPersistent which proxies requests through chained servers
-	// and does not use keep alive connections. Since no root CA is specified, we do not need to check for an error.
-
-	var httpClient *http.Client
+type serialTransport []http.RoundTripper
 
-	if baseURL == fmt.Sprintf("https://%s", common.APIBaseUrl) {
-		log.Debug("using proxied.Fronted")
-		//this is ios version
-		httpClient = &http.Client{
-			Transport: proxied.Fronted("auth_fronted_roundtrip", 30*time.Second),
+func (tr serialTransport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	for _, rt := range tr {
+		resp, err = rt.RoundTrip(req)
+		if err == nil {
+			return
 		}
-	} else {
-		log.Debug("using proxied.ChainedNonPersistent")
-		rt, _ := proxied.ChainedNonPersistent("")
-		httpClient = pro.NewHTTPClient(rt, 30*time.Second)
+		log.Debugf("Error roundtripping request to %v, continuing to next transport", req.URL)
 	}
+	log.Errorf("Unable to roundtrip request to %v, out of transports", req.URL)
+	return
+}
 
+// NewClient creates a new instance of AuthClient
+func NewClient(baseURL string, userConfig func() common.UserConfig) AuthClient {
+	chained, err := proxied.ChainedNonPersistent("")
+	if err != nil {
+		log.Fatal(err)
+	}
+	frt := proxied.Fronted("auth_fronted_roundtrip", 10*time.Second)
 	rc := webclient.NewRESTClient(&webclient.Opts{
 		BaseURL: baseURL,
 		OnBeforeRequest: func(client *resty.Client, req *http.Request) error {
 			prepareUserRequest(req, userConfig())
 			return nil
 		},
-		HttpClient: httpClient,
+		HttpClient: &http.Client{
+			Transport: serialTransport{chained, frt},
```

Comment:
This updates the auth client to use an http.Client that first attempts to connect via chained proxies and then falls back to using domain fronting with the custom name above (@jigar-f @hwh33 I'm not sure if we can just use `proxied.ChainedThenFronted()` here?).
Please let me know if this is correct otherwise feel free to change it to the one we should be using

Reply:
I remember some work needs to be done on auth API access not sure if that was done or not. , Let's wait for @hwh33 on this.

Reply:
Yeah I would just use `ChainedThenFronted`